src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpRequest.java

```diff
@@ -43,6 +44,9 @@ public NullaryExecutableOp createExecOp( final ExpectedVariables ... inputVars )
 		final ReqType req = lop.getRequest();
 		final MemberType fm = lop.getFederationMember();
 		if ( fm instanceof SPARQLEndpoint && req instanceof SPARQLRequest ) {
+			if(fm.getVocabularyMapping() != null) {
+				return new ExecOpRequestSPARQLWithTranslation( (SPARQLRequest) req, (SPARQLEndpoint) fm );
+			}
```

Comment:
No. Please remove that. This case will be covered by the new `PhysicalOpRequestWithTranslation` class.